src/ja/server/dispatcher/dispatcher.py

```diff
@@ -22,38 +25,81 @@ def __init__(self, proxy_factory: WorkerProxyFactoryBase):
         self._proxy_factory = proxy_factory
         self._previous_statuses: Dict[str, JobStatus] = dict()
 
-    def set_distribution(self, job_distribution: JobDistribution) -> None:
+    def set_distribution(self, job_distribution: JobDistribution) -> List[WorkMachine]:
         """!
         Distributes the newly added jobs to the work machines.
         and potentially pauses jobs that need to be paused.
 
         @param job_distribution the new job distribution.
+        @return a list of all work machines that the server has lost connection to.
         """
         logger.debug("Dispatching jobs")
         new_statuses: Dict[str, JobStatus] = dict()
+        self._lost_work_machines: List[WorkMachine] = []
         for job_entry in job_distribution:
             job = job_entry.job
 
             if job.status == JobStatus.QUEUED:
                 logger.debug("Job %s queued." % job.uid)
                 continue
+            if job.status == JobStatus.RUNNING:
+                logger.debug("Job %s is set to run." % job.uid)
```

Comment:
I don't think we need debug's here, since we willl print the same message twice. Better leave only the one below.

Oh, and I have other debug messages already added in the master branch, if you rebase, you can take them (I print the job, status and assigned machine which I think is better)